src/analyzer/analyzer.py

```diff
@@ -350,26 +348,26 @@ def __get_improvement(val, against, low_is_better=True):
             if "global_kernel_kernel_names" in res_proc:
                 self.__services.debug.write("Global kernel kernels: {}%".format(res_proc["global_kernel_kernel_names"]), streams=[self.__analysis_stream])
                 self.__services.debug.write("Global kernel kernel calls percentages: {}%".format(list(map(lambda r: str(r) + "%", res_proc["kernels_pick_perc"]))), streams=[self.__analysis_stream])
-        
+
         try:
             if res_proc["average_total_search_space"]:
                 pass
         except:
-            res_proc["average_total_search_space"]= float('nan')
-            res_proc['search_space_alldata']= [float('nan') for x in range(len(res_proc['path_deviation_alldata']))]
-        
+            res_proc["average_total_search_space"] = float('nan')
+            res_proc['search_space_alldata'] = [float('nan') for x in range(len(res_proc['path_deviation_alldata']))]
+
         self.__services.debug.write("\n", timestamp=False, streams=[self.__analysis_stream])
 
         # writing average data to csv file
         with open('pbtest.csv', 'a+', newline='') as file:
             fieldnames = ['Algorithm', 'Average Path Deviation', 'Success Rate', 'Average Time','Average Steps', 'Average Distance', 'Average Distance from Goal','Average Original Distance from Goal','Average Search Space', 'Average Memory']
             writer = csv.DictWriter(file, fieldnames=fieldnames)
 
-            if algorithm_type == AStar:
+            if file.tell() == 0:
```

Comment:
whats this?

Reply:
`.tell()` is the current "cursor position" of the file - 0 means it's at the beginning of the file, hence we need headers. If it's non-zero, we've written data already.